src/taker/offers.rs

```diff
@@ -160,25 +160,20 @@ pub async fn fetch_addresses_from_dns(
                     directory_server_address.as_str(),
                 )
                 .await
-                .map_err(|_e| {
-                    DirectoryServerError::Other(
-                        "Issue with fetching maker address from directory server",
-                    )
+                .map_err(|e| {
+                    DirectoryServerError::Socks5ConnectionError(format!(
+                        "Issue with fetching maker address from directory server: {}",
+                        e
+                    ))
```

Comment:
This is inside Taker, so it should return the TakerError. Previously it was a mistake to return DirectoryError here. Please change this to appropriate TakerError.